src/hooks/useNotificationModal.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  SaveSubscriptionMutation,
+  SaveSubscriptionMutationVariables,
+} from "gql/generated/types";
+import { useBannerDispatchContext } from "context/banners";
+import { useState, useEffect } from "react";
+import { useMutation } from "@apollo/react-hooks";
+import { SAVE_SUBSCRIPTION } from "gql/mutations/save-subscription";
+import get from "lodash.get";
+
+export interface UseNotificationModalProps {
+  subscriptionMethodControls: SubscriptionMethods;
+  triggers: Trigger[];
+  resourceType: ResourceType;
+  resourceId: string;
+  onCancel: () => void;
+}
+export const useNotificationModal = ({
+  triggers,
+  subscriptionMethodControls,
+  resourceType,
+  resourceId,
+  onCancel,
+}: UseNotificationModalProps) => {
+  const dispatchBanner = useBannerDispatchContext();
+  const [selectedSubscriptionMethod, setSelectedSubscriptionMethod] = useState(
+    ""
+  );
+  // target represents the input value for a subscription method
+  const [target, setTarget] = useState<Target>({});
+  const [isFormValid, setIsFormValid] = useState(false);
+  const [extraFieldErrorMessages, setExtraFieldErrorMessages] = useState<
+    string[]
+  >([]);
+  const [saveSubscription, { loading: mutationLoading }] = useMutation<
+    SaveSubscriptionMutation,
+    SaveSubscriptionMutationVariables
+  >(SAVE_SUBSCRIPTION, {
+    onCompleted: () => {
+      dispatchBanner.successBanner("Your subscription has been added");
+    },
+    onError: (err) => {
+      dispatchBanner.errorBanner(
+        `Error adding your subscription: '${err.message}'`
+      );
```

Comment:
Its a little odd to have the banner toggled from inside a custom hook. I think i would prefer if it was called from outside the hook at the component level. What do you think?

Reply:
i can see what you mean... but is also depends on the hook and how its used.